uniffi_bindgen/src/interface/enum_.rs

```diff
@@ -78,65 +78,84 @@
 
 use anyhow::{bail, Result};
 
-use super::record::Field;
+use super::record::{Field, FieldDescr};
 use super::types::Type;
-use super::{APIConverter, ComponentInterface};
+use super::{APIConverter, ComponentInterface, CINode};
 
 /// Represents an enum with named variants, each of which may have named
 /// and typed fields.
 ///
 /// Enums are passed across the FFI by serializing to a bytebuffer, with a
 /// i32 indicating the variant followed by the serialization of each field.
+#[derive(Debug)]
+pub struct Enum<'a> {
+    pub(super) parent: &'a ComponentInterface,
+    pub(super) descr: &'a EnumDescr,
+}
+
 #[derive(Debug, Clone, Hash)]
-pub struct Enum {
+pub struct EnumDescr {
     pub(super) name: String,
-    pub(super) variants: Vec<Variant>,
+    pub(super) variants: Vec<VariantDescr>,
     // "Flat" enums do not have, and will never have, variants with associated data.
     pub(super) flat: bool,
 }
 
-impl Enum {
+impl<'a> Enum<'a> {
     pub fn name(&self) -> &str {
-        &self.name
+        &self.descr.name
     }
 
-    pub fn variants(&self) -> Vec<&Variant> {
-        self.variants.iter().collect()
+    pub fn variants(&self) -> Vec<Variant<'_, Self>> {
```

Comment:
Here, when consuming code lists the variants of the enum, so see that we create a bunch of `Variant` instances on demand with the appropriate pointers.